pkg/build/strategies/layered/layered.go

```diff
@@ -139,8 +167,14 @@ func (b *Layered) Build(config *api.Config) (*api.Result, error) {
 	b.config.LayeredBuild = true
 	// new image name
 	b.config.BuilderImage = newBuilderImage
-	// the scripts are inside the image
-	b.config.ScriptsURL = "image://" + filepath.Join(getDestination(config), "scripts")
+	// see CreateDockerfile, conditional copy, location of scripts
+	scriptsIncluded := scriptsProvided(filepath.Join(config.WorkingDir, api.UploadScripts))
+	glog.V(2).Infof("Scripts dir has contents %v", scriptsIncluded)
+	if scriptsIncluded {
+		b.config.ScriptsURL = "image://" + filepath.Join(getDestination(config), "scripts")
+	} else {
+		b.config.ScriptsURL = os.Getenv("STI_SCRIPTS_PATH")
```

Comment:
STI_SCRIPTS_PATH is an image-specific implementation detail/convenience.  the API for defining where the scripts exist within the image is the label:
https://github.com/openshift/sti-base/blob/master/Dockerfile#L8

I think we probably need to sit down for 10-15 minutes and actually define what the flows look like when one or more scripts come from the various potential locations.

In an ideal implementation, "scriptsIncluded" would always be true because no matter where we got them from (a user provided external url, the user's source repo, a user provided image:// url, or a url defined by the image itself via the aforementioned label) we would always go fetch them from there and copy them into the "scripts" directory in the working directory (peer to the source directory) and then run.

if no scripts are found at that point, it's an error because none were provided.

the key point is that:
https://github.com/openshift/sti-base/blob/master/Dockerfile#L8

is really just another url (it would be valid for that to be an http url), it's just a URL with the lowest/worst priority, with the files at that URL being used only if the user didn't explicilty specify a URL or provide scripts in their source dir.